packages/core/conductor-tasks/bichard_phase_2/processPhase2.ts

```diff
@@ -32,12 +43,13 @@ const processPhase2: ConductorWorker = {
       return failed(`Could not put file to S3: ${s3TaskDataPath}`, s3PutResult.message)
     }
 
-    const hasTriggersOrExceptions =
-      ("triggers" in result && result.triggers.length > 0) ||
-      ("hearingOutcome" in result && result.outputMessage.Exceptions.length > 0)
+    const hasTriggersOrExceptionsOrIgnored =
+      result.triggersGenerated ||
+      result.outputMessage.Exceptions.length > 0 ||
+      result.resultType === Phase2ResultType.ignored
 
```

Comment:
We need to persist phase 2 result, when
- `triggersGenerated` is true. That means the function to generate triggers has been called in Phase 2.
- There are any exceptions
- `resultType` is `ignored`. We also update the record when the case is ignored as it is possible that the record is resubmitted to Phase 1, and the error status is `Submitted`. So if case is ignored in Phase 2, we need to update the error_status in the database.

Reply:
For additional context on `triggersGenerated` from our Slack convo:

>If we don't use `triggersGenerated` and only rely on the `triggers` array, then `updateErrorListTriggers` will delete existing triggers because it compares existing triggers with the `triggers` array and delete the ones that don't exist in the `triggers` array.